go/vt/tabletserver/query_executor.go

```diff
@@ -295,25 +295,38 @@ func (qre *QueryExecutor) execDDL() (*sqltypes.Result, error) {
 		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "DDL is not understood")
 	}
 
+	if qre.transactionID != 0 {
```

Comment:
Liquibase disables autocommit so that it can handle some changesets such as adding a not-null constraint with a default value. This turns into 2 queries: 1 to update the existing values, and 1 to add the constraint. These should happen in a single transaction.  Liquibase disables setAutoCommit, so the JDBC does a BEGIN to start the transaction. This works through vtgate, but when vttablet sees it it starts its own transaction. The two transactions actually deadlock each other in this case.

The downside of this change is that for these multi-query transactions we no longer call `schema.Engine.Reload()`. Trying to think of ways around that.